cmp/grid/impl/ColumnHeader.js

```diff
@@ -75,8 +75,8 @@ export const columnHeader = hoistCmp.factory({
             return div({
                 className: 'xh-grid-header-expand-collapse-icon',
                 item: icon,
-                onClick: isDesktop ? model.onExpandOrCollapse : null,
-                onTouchStart: !isDesktop ? model.onExpandOrCollapse : null
+                onClick: hasMouseSupport ? model.onExpandOrCollapse : null,
+                onTouchStart: !hasMouseSupport ? model.onExpandOrCollapse : null
```

Comment:
Doesn't this mean that tablets no longer respond to touch events? Or do iPads fire both kind of events when interacted with via the touch screen?

Reply:
Seems tablets fire click events 300ms after touchStart, to make sure you keep your finger on the target and weren't simply touching a target to initiate a swipe (to scroll) or double-tap (to zoom).  I was able to run this on my iPad and get the desired behavior touching and mouse-clicking.

Here's a couple articles I found:
https://www.quirksmode.org/blog/archives/2014/04/suppressing_the.html
https://www.sitepoint.com/5-ways-prevent-300ms-click-delay-mobile-devices/

Another approach could be to bind both events and add preventDefault() to onTouchStart since that fires first.

In the future, may be able to use this proposed PointerEvent interface:
https://www.w3.org/TR/pointerevents/
"To reduce the cost of coding to multiple input types and also to help with the above described ambiguity with Mouse Events, this specifications defines a more abstract form of input, called a [pointer](https://www.w3.org/TR/pointerevents/#dfn-pointer). A pointer can be any point of contact on the screen made by a mouse cursor, pen, touch (including multi-touch), or other pointing input device."

Reply:
I wouldn't hold my breath for PointerEvents :)

I'd like to see if we can try the approach of binding both events and using `preventDefault()` on touch events. Then we can favour touch events, and keep click events as a fallback. Maybe even losing the conditional binding consistently.

It just seems strange to me route touch events through the click event methods on tablets in order to support a trackpad (i.e. a 'secondary' input method)

Reply:
Makes sense